core/services/gateway/handlers/functions/allowlist.go

```diff
@@ -191,3 +232,31 @@ func (a *onchainAllowlist) update(addrList []common.Address) {
 	a.allowlist.Store(&newAllowlist)
 	a.lggr.Infow("allowlist updated successfully", "len", len(addrList))
 }
+
+func (a *onchainAllowlist) updateCache(addrList []common.Address) {
+	for id, addr := range addrList {
+		if err := a.orm.UpsertAllowedSender(uint64(id), addr); err != nil {
+			a.lggr.Errorf("failed to update cache: %w", err)
+		}
+	}
+}
+
+func (a *onchainAllowlist) loadCachedAllowedList() {
+	allowedList := make([]common.Address, 0)
+	offset := uint(0)
+	for {
+		asBatch, err := a.orm.GetAllowedSenders(offset, a.config.CacheBatchSize)
```

Comment:
Is it worth decoupling the config of batch size for on-chain queries versus the batch size for DB queries?

Reply:
makes sense we might not want to fetch same amount from onchain as from db. Added 👍🏼